src/Microsoft.AspNet.Mvc.Core/Controllers/DefaultControllerActionArgumentBinder.cs

```diff
@@ -145,22 +157,31 @@ private static void CallPropertyAddRange<TElement>(object target, object source)
             }
         }
 
-        private void ActivateProperties(object controller, Type containerType, Dictionary<string, object> properties)
+        private void ActivateProperties(
+            ControllerActionDescriptor actionDescriptor,
+            object controller,
+            IDictionary<string, object> properties)
         {
             var propertyHelpers = PropertyHelper.GetProperties(controller);
-            foreach (var property in properties)
+            foreach (var property in actionDescriptor.BoundProperties)
```

Comment:
Rejiggered this to iterate over the `BoundProperties` which is an array, rather than the `IDictionary<>` which would allocate.